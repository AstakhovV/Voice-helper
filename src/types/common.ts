export interface Commands {
  command: string;
  callback: () => void;
}
