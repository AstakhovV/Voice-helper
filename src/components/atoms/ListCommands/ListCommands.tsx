import React from 'react';
import { Command } from '../../../types/common';

interface Props {
  commands: Command[];
}

const ListCommands = ({ commands }: Props) => (
  <ul className="flex flex-col gap-2">
    {commands.map((command) => (
      <li key={command.id} className="flex gap-4 rounded bg-slate-500 p-2">
        {command.command} - {command.description}
      </li>
    ))}
  </ul>
);

export default ListCommands;
