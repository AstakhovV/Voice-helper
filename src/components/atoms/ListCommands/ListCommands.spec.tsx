import React from 'react';
import { render, screen } from '@testing-library/react';
import ListCommands from './ListCommands';
import { Command } from '../../../types/common';

const commands: Command[] = [
  {
    id: '12312',
    command: 'Render',
    description: 'Render something',
    callback: () => console.log('render'),
  },
];

describe('Render Navbar', () => {
  beforeAll(() => {
    render(<ListCommands commands={commands} />);
  });

  test('Navbar renders correctly', () => {
    const list = screen.getByRole('list');
    expect(list).toMatchSnapshot();
  });

  test('Command renders correctly', () => {
    const listItems = screen.getAllByRole('listitem');
    for (const element of listItems) {
      expect(element).toBeInTheDocument();
    }
    expect(listItems[0]).toHaveTextContent('Render - Render something');
  });
});
