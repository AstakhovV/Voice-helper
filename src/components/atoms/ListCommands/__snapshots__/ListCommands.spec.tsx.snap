// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navbar Navbar renders correctly 1`] = `
<ul
  class="flex flex-col gap-2"
  role="list"
>
  <li
    class="flex gap-4 rounded bg-slate-500 p-2"
  >
    Render
     - 
    Render something
  </li>
</ul>
`;
